### PR TITLE
code refactoring

### PR DESCRIPTION
**Pull request by nelly-hateva**

### CODE REVIEW

File: src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    private void extractJaroWinklerNamesFeatures() {
+        sparseVector.add(xA.lookupObject("jwf"), JaroWinkler.distance(person1.firstName, person2.firstName));
+        sparseVector.add(xA.lookupObject("jwl"), JaroWinkler.distance(person1.lastName, person2.lastName));
+
+        sparseVector.add(xA.lookupObject("jw"), JaroWinkler.distance(person1.firstName + " " + person1.lastName,
+                person2.firstName + " " + person2.lastName));
+    }
+
+    private void extractJaroWinklerNormalizedNamesFeatures() {
+        sparseVector.add(xA.lookupObject("jwnf"), JaroWinkler.distance(person1.normalizedFirstName, person2.normalizedFirstName));
```
Comment (ivelinanikolova) [inline src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java]: Do you think that just taking the distance is a good feature, maybe discretizing it to bins will do a better job for classification?

File: src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        Set<String> secondPersonNormalizedNameDaitchMokotoffEncodingsSet = new HashSet<>(Arrays.asList(
+                daitchMokotoffSoundex.soundex(person2.normalizedName).split("|")
+        ));
+        firstPersonNormalizedNameDaitchMokotoffEncodingsSet.retainAll(secondPersonNormalizedNameDaitchMokotoffEncodingsSet);
+
+        sparseVector.add(xA.lookupObject("dms"), (firstPersonNormalizedNameDaitchMokotoffEncodingsSet.size() == 0) ? 0.0d : 1.0d);
+    }
+
+    private void extractMotherNameFeatures() {
+        sparseVector.add(xA.lookupObject("jwnm"), JaroWinkler.distance(
```
Comment (ivelinanikolova) [inline src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java]: Same comment as above about discretization of the distance.

File: src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                person1.nameMotherFirstName + " " + person1.nameMotherLastName,
+                person1.nameMotherFirstName + " " + person1.nameMotherLastName
+        ));
+    }
+
+    private void extractPlaceBirthFeatures() {
+        sparseVector.add(xA.lookupObject("lpb"), Levenshtein.similarity(person1.placeBirth, person2.placeBirth));
```
Comment (ivelinanikolova) [inline src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java]: Same comment about discretization of the distance.

File: src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    private void extractDateBirthFeatures() {
```
Comment (ivelinanikolova) [inline src/main/java/com/ontotext/ehri/deduplication/model/USHMMClassificationInstance.java]: Same comment about discretization of the distance.

File: src/main/java/com/ontotext/ehri/deduplication/model/USHMMPersonsFeatureExtractor.java [hunk 5]
```diff
@@ -21,196 +13,22 @@
...
-                (statementsMap.get(personId1, "personType").equals(statementsMap.get(personId2, "personType"))) ? 0.0d : 1.0d);
-    }
+    private ClassificationInstance getInstance(USHMMPersonStatementsMapHash statementsMap, USHMMGoldStandardEntry entry, Alphabet xA, Alphabet yA) {
+        USHMMPersonPair personPair = new USHMMPersonPair(entry, statementsMap);
+        USHMMPerson person1 = personPair.getPerson1();
+        USHMMPerson person2 = personPair.getPerson2();
+        USHMMClassificationInstance instance = new USHMMClassificationInstance(xA, person1, person2);
+        return new ClassificationInstance(xA, yA, instance.getSparseVector(), yA.lookupObject(entry.getLabel()));
 
-    private void extractOccupationFeature(Alphabet xA, String personId1, String personId2, SparseVector sparseVector) {
```
Comment (ivelinanikolova) [inline src/main/java/com/ontotext/ehri/deduplication/model/USHMMPersonsFeatureExtractor.java]: I didn't notice occupation feature elsewhere, it's good to have it implemented even if it is not used in the current mode.

Comment (nelly-hateva) [inline src/main/java/com/ontotext/ehri/deduplication/model/USHMMPersonsFeatureExtractor.java]: It is used in the current mode, line 157 :)

**Pull request merged**